medusa/providers/torrent/html/privatehd.py

```diff
@@ -0,0 +1,184 @@
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+                print("Search string:" + search_string)
+
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                search_params = {
+                    'in': 1,
+                    'search': search_string,
+                    'type': 2
+                }
+
+                if not search_string:
+                    del search_params['search']
+
+                response = self.session.get(self.urls['search'], params=search_params)
+                if not response or not response.text:
+                    log.debug('No data returned from provider')
+                    continue
+
+                results += self.parse(response.text, mode)
+
+        return results
+
+    def parse(self, data, mode):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            if not html:
+                log.debug('No html data parsed from provider')
+                return items
+
+            torrents = html('tr')
+            if not torrents or len(torrents) < 2:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            # Skip column headers
+            for row in torrents[1:]:
+                # Skip extraneous rows at the end
+                if len(row.contents) < 10:
+                    continue
+
+                try:
+                    title = row.find(class_='torrent-filename').text.strip()
+                    download_url = row.find(class_='torrent-download-icon').get('href')
+                    size = convert_size(row.contents[11].text.strip(), -1)
+                    seeders = row.contents[13].text
+                    leechers = row.contents[15].text
+                    pubdate = self.parse_pubdate(row.contents[7].findChild().get('title'))
+
+                    item = {
+                        'title': title,
+                        'link': download_url,
+                        'size': size,
+                        'seeders': seeders,
+                        'leechers': leechers,
+                        'pubdate': pubdate,
+                    }
+                    if mode != 'RSS':
+                        pass
+                    log.debug('Found result: {0} with {1} seeders and {2} leechers',
+                              title, seeders, leechers)
+
+                    items.append(item)
+                except (AttributeError, TypeError, KeyError, ValueError, IndexError):
+                    log.error('Failed parsing provider. Traceback: {0!r}',
```

Comment:
Use log.exception here instead

Reply:
log.exception('Failed parsing provider.')

Reply:
It seems that the majority of providers use log.error, or is log.exception the new way?

Reply:
Yes, we change it recently.